CMakeLists.txt

```diff
@@ -1,198 +1,255 @@
-  ${catkin_LIBRARIES} ${PROJECT_NAME} ${PROJECT_NAME}_imarker_simple
-)
-set_target_properties(${PROJECT_NAME}_imarker_simple_demo
-  PROPERTIES
-  OUTPUT_NAME
-  imarker_simple_demo PREFIX ""
+  ${PROJECT_NAME}_imarker_simple
 )
 
-##########
-## TEST ##
-##########
-
-if(CATKIN_ENABLE_TESTING)
-
-  add_rostest_gtest(rvt_test launch/rvt_test.launch tests/rvt_test.cpp)
-  target_link_libraries(rvt_test
-    ${PROJECT_NAME}
-    ${catkin_LIBRARIES}
-    ${Boost_LIBRARIES}
-  )
-
-endif()
-
-## Test for correct C++ source code
-roslint_cpp()
+ament_target_dependencies(${PROJECT_NAME}_imarker_simple_demo
+  rclcpp
+)
 
 #############
 ## Install ##
 #############
 
+ament_export_libraries(${PROJECT_NAME})
+ament_export_libraries(${PROJECT_NAME}_remote_control)
+# ament_export_libraries(${PROJECT_NAME}_gui)
+ament_export_libraries(${PROJECT_NAME}_imarker_simple)
+
 # Install libraries
 install(
   TARGETS
     ${PROJECT_NAME}
-    ${PROJECT_NAME}_gui
-    ${PROJECT_NAME}_imarker_simple
     ${PROJECT_NAME}_remote_control
-  LIBRARY DESTINATION
-    ${CATKIN_PACKAGE_LIB_DESTINATION}
-  ARCHIVE DESTINATION
-    ${CATKIN_PACKAGE_LIB_DESTINATION}
-  RUNTIME DESTINATION
-    ${CATKIN_GLOBAL_BIN_DESTINATION}
+    # ${PROJECT_NAME}_gui
+    ${PROJECT_NAME}_imarker_simple
+  LIBRARY DESTINATION lib
+  ARCHIVE DESTINATION lib
+  RUNTIME DESTINATION bin
 )
 
-# Install header files
-install(DIRECTORY include/${PROJECT_NAME}/   DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION})
+# Install executables
+install(
+  TARGETS
+    ${PROJECT_NAME}_demo
+    ${PROJECT_NAME}_imarker_simple_demo
+  LIBRARY DESTINATION lib
+  ARCHIVE DESTINATION lib
+  RUNTIME DESTINATION lib/${PROJECT_NAME}
+)
+
+install(
+  DIRECTORY "${CMAKE_SOURCE_DIR}/meshes"
+  DESTINATION "share/${PROJECT_NAME}"
+  )
 
 # Install shared resources
-install(DIRECTORY launch    DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION})
-install(DIRECTORY resources DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION})
+install(DIRECTORY launch    DESTINATION share/${PROJECT_NAME})
+# install(DIRECTORY meshes    DESTINATION share/${PROJECT_NAME})
 
 # Install xml plugin config
-install(FILES
-  plugin_description.xml
-  DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}
-)
+# install(FILES
+#   plugin_description.xml
+#   DESTINATION share/${PROJECT_NAME}
+# )
 
-# Install executables
-install(TARGETS ${PROJECT_NAME}_demo ${PROJECT_NAME}_imarker_simple_demo
-  LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-  ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-  RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
+# Install header files
+install(
+  DIRECTORY include/
+  DESTINATION include)
+
+##########
+## TEST ##
+##########
+
+if (BUILD_TESTING)
+  # # Lint
+  # find_package(ament_lint_auto REQUIRED)
+  # ament_lint_auto_find_test_dependencies()
+
+  # # Run tests
+  # find_package(ament_cmake_gtest REQUIRED)
+  # ament_add_gtest(${PROJECT_NAME}_rvt_test tests/rvt_test.cpp)
```

Comment:
Do not disable tests! I spent a lot of time on this and its how you verify your port works...